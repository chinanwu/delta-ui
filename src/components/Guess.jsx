import PropTypes from 'prop-types';
import React, { useCallback, useEffect, useRef, useState } from 'react';
import { connect } from 'react-redux';

import {
	aBtn,
	backspaceBtn,
	enterBtn,
	leftArrowBtn,
	rightArrowBtn,
	tabBtn,
	zBtn,
} from '../constants/Keycodes';
import getThemeClassname from '../functions/getThemeClassname';

import './Guess.less';

const maxLen = 4;
const regex = /([a-zA-Z])/g;

export const Guess = ({ dark, prevWord, error, onGuess }) => {
	const [guessVals, setGuessVals] = useState(['.', '.', '.', '.']);
	const inputRefs = [useRef(null), useRef(null), useRef(null), useRef(null)];

	useEffect(() => {
		setGuessVals(prevWord.match(regex));
	}, [prevWord]);

	const handleKeyDown = useCallback(
		event => {
			if (
				event &&
				event.target &&
				event.keyCode &&
				event.key &&
				!event.ctrlKey &&
				!event.altKey &&
				!event.metaKey
			) {
				const i = parseInt(event.target.dataset.id);
				const next = (i + 1) % maxLen;
				const prev = i - 1 < 0 ? maxLen - 1 : i - 1;

				if (event.keyCode >= aBtn && event.keyCode <= zBtn) {
					event.preventDefault();
					inputRefs[i].current.value = event.key;
					if (next >= i) {
						inputRefs[next].current.focus();
						inputRefs[next].current.select();
					}
				} else {
					switch (event.keyCode) {
						case enterBtn:
							event.preventDefault();
							handleEnterClick();
							inputRefs[0].current.focus();
							break;
						case backspaceBtn:
							event.preventDefault();
							inputRefs[i].current.value = '';
							if (prev <= i) {
								inputRefs[prev].current.focus();
								inputRefs[prev].current.select();
							}
							break;
						case rightArrowBtn:
							inputRefs[next].current.focus();
							inputRefs[next].current.select();
							break;
						case leftArrowBtn:
							inputRefs[prev].current.focus();
							inputRefs[prev].current.select();
							break;
						case tabBtn:
							//do nothing
							break;
						default:
							event.preventDefault();
							break;
					}
				}
			}
		},
		[inputRefs]
	);

	const handleEnterClick = useCallback(() => {
		let guess = '';
		inputRefs.forEach((inputRef, i) => {
			if (inputRef.current.value !== '') {
				guess = guess + inputRef.current.value;
				inputRef.current.value = '';
			} else {
				guess = guess + prevWord[i];
			}
		});

		onGuess(guess);
	}, [inputRefs, onGuess]);

	return (
		<div className="Guess">
			<h3 id="guessLabel" className="Guess__label">
				Next Word:
			</h3>
			<div className="Guess__inputs">
				<div>
					{guessVals &&
						guessVals.map((val, i) => (
							<input
								id={'guessInput_' + i}
								ref={inputRefs[i]}
								data-id={i}
								className={getThemeClassname('Guess__input', dark)}
								type="text"
								maxLength={1}
								placeholder={val}
								aria-labelledby="guessLabel"
								aria-invalid={error}
								onKeyDown={handleKeyDown}
								key={'guessInput-' + i}
							/>
						))}
				</div>

				<button
					id="guessEnterBtn"
					className="Guess__enterBtn"
					onClick={handleEnterClick}
				>
					Enter
				</button>
			</div>

			<div
				className={getThemeClassname('Guess__error', dark)}
				aria-live="passive"
			>
				{error}
			</div>
		</div>
	);
};

export const mapStateToProps = ({ theme: { dark } }) => ({
	dark,
});

Guess.propTypes = {
	dark: PropTypes.bool,
	prevWord: PropTypes.string,
	onGuess: PropTypes.func,
};

export default connect(mapStateToProps)(Guess);
