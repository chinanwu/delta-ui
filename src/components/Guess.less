@import '../style/buttons';

.Guess {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	width: 30rem;
}

.Guess__label {
	margin-bottom: 2rem;
}

.Guess__inputs {
	display: flex;
	justify-content: center;
	position: absolute;
}

.Guess__input {
	margin: 0.4rem;
	width: 3.6rem;
	height: 4.2rem;
	text-align: center;
	font-size: 1.6rem;

	&::-moz-placeholder {
		// Make placeholder text colour AAA
		color: #595959;
	}

	&:-ms-input-placeholder {
		color: #595959;
	}

	&::-webkit-input-placeholder {
		color: #595959;
	}
}

.Guess__input--dark {
	color: white;
	background: @dark-02;
	border: 1px solid black;
	padding: 0.3rem;
	&::-moz-placeholder {
		// Make placeholder text colour AAA
		color: #bfbfbf;
	}

	&:-ms-input-placeholder {
		color: #bfbfbf;
	}

	&::-webkit-input-placeholder {
		color: #bfbfbf;
	}
}

.Guess__enterBtn {
	@smBtn();
}

.Guess__error {
	font-size: 1.8rem;
	text-align: center;
	color: @error;
	margin-top: 1rem;
	position: relative;
	top: 3rem;
	height: 6rem;
}

.Game__error--dark {
	color: @errorDark;
}

@media only screen and (max-width: 550px) {
	.Guess {
		width: initial;
	}

	.Guess__label {
		margin-bottom: 0;
	}

	.Guess__inputs {
		position: initial;
	}

	.Guess__enterBtn {
		font-size: 125%;
	}

	.Guess__error {
		position: initial;
		top: initial;
	}
}

@media only screen and (max-width: 375px) {
	.Guess__input {
		font-size: 2rem;
	}
}

@media only screen and (max-width: 310px) {
	.Guess__inputs {
		flex-direction: column;
	}
}
