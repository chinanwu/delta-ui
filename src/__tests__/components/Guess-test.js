import React, { useEffect, useState, useRef } from 'react';
import { shallow } from 'enzyme';

import { Guess, mapStateToProps } from '../../components/Guess.jsx';
import {
	aBtn,
	backspaceBtn,
	enterBtn,
	escapeBtn,
	leftArrowBtn,
	rightArrowBtn,
	tabBtn,
	zBtn,
} from '../../constants/Keycodes';

jest.unmock('../../components/Guess.jsx');

describe('Guess', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders with props', () => {
			const setGuessVals = jest.fn();
			useState.mockReturnValue([['.', '.', '.', '.'], setGuessVals]);
			const wrapper = shallow(
				<Guess dark={true} prevWord="food" error="" onGuess={jest.fn()} />
			);
			expect(wrapper.getElement()).toMatchSnapshot();
			useEffect.mock.calls[0][0]();
			expect(setGuessVals.mock.calls).toEqual([[['f', 'o', 'o', 'd']]]);
		});
	});

	describe('behaviour', () => {
		it('submits when enter button is clicked', () => {
			const onGuess = jest.fn();
			const setGuessVals = jest.fn();
			const ref = { current: { value: 'f' } };
			useState.mockReturnValue([['g', 'o', 'o', 'd'], setGuessVals]);
			useRef.mockReturnValue(ref);
			const wrapper = shallow(
				<Guess dark={true} prevWord="food" error="" onGuess={onGuess} />
			);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessEnterBtn').simulate('click');
			expect(onGuess.mock.calls).toEqual([['good']]);
			expect(setGuessVals.mock.calls).toEqual([[['f', 'o', 'o', 'd']]]);
		});

		it('submits when enter is pressed', () => {
			const preventDefault = jest.fn();
			const onGuess = jest.fn();
			const setGuessVals = jest.fn();
			const focus = jest.fn();
			const ref = { current: { value: 'f', focus } };
			const event = {
				target: { dataset: { id: 0 } },
				key: 'enter',
				keyCode: enterBtn,
				preventDefault: preventDefault,
			};
			useState.mockReturnValue([['g', 'o', 'o', 'd'], setGuessVals]);
			useRef.mockReturnValue(ref);
			const wrapper = shallow(<Guess prevWord="food" onGuess={onGuess} />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(onGuess.mock.calls).toEqual([['good']]);
			expect(setGuessVals.mock.calls).toEqual([[['f', 'o', 'o', 'd']]]);
			expect(focus).toHaveBeenCalledTimes(1);
			expect(preventDefault).toHaveBeenCalledTimes(1);
		});

		it('moves to next box when enter a letter', () => {
			const setGuessVals = jest.fn();
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'g' } };
			const secondRef = { current: { value: 'o', focus, select } };
			const otherRefs = { current: { value: 'o' } };
			const event = {
				target: { dataset: { id: 0 } },
				key: 'a',
				keyCode: aBtn,
				preventDefault: preventDefault,
			};
			useState.mockReturnValue([['g', 'o', 'o', 'd'], setGuessVals]);
			useRef
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(secondRef)
				.mockReturnValue(otherRefs);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(focus).toHaveBeenCalledTimes(1);
			expect(select).toHaveBeenCalledTimes(1);
			// expect(setGuessVals.mock.calls).toEqual([[['a', 'o', 'o', 'd']]]); // TODO: Functional state setter
			expect(setGuessVals).toHaveBeenCalledTimes(1);
			expect(preventDefault).toHaveBeenCalledTimes(1);
			expect(ref.current.value).toEqual('a');
		});

		it("doesn't move to first box when enter last letter", () => {
			const setGuessVals = jest.fn();
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'o', focus, select } };
			const currentRef = { current: { value: 'd', focus, select } };
			const event = {
				target: { dataset: { id: 3 } },
				key: 'z',
				keyCode: zBtn,
				preventDefault: preventDefault,
			};
			useState.mockReturnValue([['g', 'o', 'o', 'd'], setGuessVals]);
			useRef
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(currentRef);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_3').simulate('keydown', event);
			expect(focus).not.toHaveBeenCalled();
			expect(select).not.toHaveBeenCalled();
			// expect(setGuessVals.mock.calls).toEqual([[['a', 'o', 'o', 'd']]]); // TODO: Functional state setter
			expect(setGuessVals).toHaveBeenCalledTimes(1);
			expect(preventDefault).toHaveBeenCalledTimes(1);
			expect(currentRef.current.value).toEqual('z');
		});

		it('goes to previous box and focuses when backspace', () => {
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'f', focus, select } };
			const currentRef = { current: { value: 'o' } };
			const otherRefs = { current: { value: 'o' } };
			const event = {
				target: { dataset: { id: 1 } },
				key: 'backspace',
				keyCode: backspaceBtn,
				preventDefault: preventDefault,
			};
			useRef
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(currentRef)
				.mockReturnValue(otherRefs);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_1').simulate('keydown', event);
			expect(focus).toHaveBeenCalledTimes(1);
			expect(select).toHaveBeenCalledTimes(1);
			expect(preventDefault).toHaveBeenCalledTimes(1);
			expect(currentRef.current.value).toEqual('');
		});

		it("doesn't do anything when backspace on first box", () => {
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'f', focus, select } };
			const event = {
				target: { dataset: { id: 0 } },
				key: 'backspace',
				keyCode: backspaceBtn,
				preventDefault: preventDefault,
			};
			useRef.mockReturnValue(ref);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_1').simulate('keydown', event);
			expect(focus).not.toHaveBeenCalled();
			expect(select).not.toHaveBeenCalled();
			expect(preventDefault).toHaveBeenCalledTimes(1);
			expect(ref.current.value).toEqual('');
		});

		it('moves to next box when right arrow', () => {
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'f' } };
			const secondRef = { current: { value: 'f', focus, select } };
			const event = {
				target: { dataset: { id: 0 } },
				key: 'right arrow',
				keyCode: rightArrowBtn,
				preventDefault: preventDefault,
			};
			useRef
				.mockReturnValueOnce(ref)
				.mockReturnValueOnce(secondRef)
				.mockReturnValue(ref);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(focus).toHaveBeenCalledTimes(1);
			expect(select).toHaveBeenCalledTimes(1);
			expect(preventDefault).not.toHaveBeenCalled();
		});

		it('moves to previous box when left arrow', () => {
			const preventDefault = jest.fn();
			const focus = jest.fn();
			const select = jest.fn();
			const ref = { current: { value: 'f', focus, select } };
			const otherRefs = { current: { value: 'f' } };
			const event = {
				target: { dataset: { id: 1 } },
				key: 'left arrow',
				keyCode: leftArrowBtn,
				preventDefault: preventDefault,
			};
			useRef.mockReturnValueOnce(ref).mockReturnValue(otherRefs);
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(focus).toHaveBeenCalledTimes(1);
			expect(select).toHaveBeenCalledTimes(1);
			expect(preventDefault).not.toHaveBeenCalled();
		});

		it('does nothing when tab is pressed', () => {
			const preventDefault = jest.fn();
			const event = {
				target: { dataset: { id: 0 } },
				key: 'tab',
				keyCode: tabBtn,
				preventDefault: preventDefault,
			};
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(preventDefault).not.toHaveBeenCalled();
		});

		it('prevents other key presses', () => {
			const preventDefault = jest.fn();
			const event = {
				target: { dataset: { id: 0 } },
				key: 'escape',
				keyCode: escapeBtn,
				preventDefault: preventDefault,
			};
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(preventDefault).toHaveBeenCalledTimes(1);
		});

		it('does nothing when event is invalid', () => {
			const preventDefault = jest.fn();
			const event = {
				target: { dataset: { id: 0 } },
				key: 'backspace',
				keyCode: backspaceBtn,
				preventDefault: preventDefault,
				ctrlKey: true,
			};
			const wrapper = shallow(<Guess />);
			expect(wrapper.getElement()).toMatchSnapshot();
			wrapper.find('#guessInput_0').simulate('keydown', event);
			expect(preventDefault).not.toHaveBeenCalled();
		});
	});

	describe('mapStateToProps', () => {
		it('returns mapped properties', () => {
			const theme = { dark: true };
			expect(mapStateToProps({ theme })).toEqual(theme);
		});
	});
});
