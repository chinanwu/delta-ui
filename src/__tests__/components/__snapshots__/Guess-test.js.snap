// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guess behaviour does nothing when event is invalid 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour does nothing when tab is pressed 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour doesn't do anything when backspace on first box 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour doesn't move to first box when enter last letter 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour goes to previous box and focuses when backspace 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour moves to next box when enter a letter 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour moves to next box when right arrow 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour moves to previous box when left arrow 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour prevents other key presses 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess behaviour submits when enter button is clicked 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  >
    
  </div>
</div>
`;

exports[`Guess behaviour submits when enter is pressed 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="g"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="o"
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="d"
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess rendering renders expected elements 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  />
</div>
`;

exports[`Guess rendering renders with props 1`] = `
<div
  className="Guess"
>
  <h3
    className="Guess__label"
    id="guessLabel"
  >
    Next Word:
  </h3>
  <div
    className="Guess__inputs"
  >
    <div>
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={0}
        id="guessInput_0"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={1}
        id="guessInput_1"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={2}
        id="guessInput_2"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
      <input
        aria-invalid=""
        aria-labelledby="guessLabel"
        data-id={3}
        id="guessInput_3"
        maxLength={1}
        onKeyDown={[Function]}
        placeholder="."
        type="text"
      />
    </div>
    <button
      className="Guess__enterBtn"
      id="guessEnterBtn"
      onClick={[Function]}
    >
      Enter
    </button>
  </div>
  <div
    aria-live="passive"
  >
    
  </div>
</div>
`;
